controller/api/destination/endpoint_translator.go

```diff
@@ -102,9 +116,11 @@ func (et *endpointTranslator) sendFilteredUpdate(set watcher.AddressSet) {
 
 	if len(diffAdd.Addresses) > 0 {
 		et.sendClientAdd(diffAdd)
+		et.watchEndpointPolicy(diffAdd)
```

Comment:
When we subscribe, we'll hit this with a set of addresses and send one big Add message containing all the endpoints in the service.  Then we'll subscribe to the policy for each of the endpoints and send an additional individual Add for each endpoint.  (and then continue to send an individual Add every time policy changes).

I'm not sure exactly how we'd want this to work, but this seems redundant and inefficient.  Perhaps we need some kind of buffering strategy to group the individual Adds into a single message?  Or maybe we can keep them as-is but instead forgo the initial Add of the entire address set?

Reply:
I've removed this "multiphase" updating in recent commits. `sendClientAdd` now only sends add updates when dealing with addresses not backed by pods. `watchEndpointPolicy` only sends add updates for addresses backed by pods. This does mean that that adds for pods now occur individually instead of grouped together, but the first destination profile returned does now take into account server resources.